### DISCUSSION TITLE
Django-cotton-like html tag syntax for django-components

### DISCUSSION THREAD

**Discussion by oliverhaas** [django-components/django-components#792] [Show and tell]
I (and one of my employees) tinkered with adding django-cotton-like syntax to django-components, because I initially loved the django-cotton syntax, but also wanted some of the functionality of django-components that django-cotton lacks. 

https://gist.github.com/oliverhaas/49a1683c6a0a4d5bac107dbfd72b283c

I would say that this is more than a POC, but it does not all the reasonable features it in principle could.

There are some upsides to the syntax (nicer formatting and normal html css class autocomplete for example), but I'm not sure anymore if the syntax is really that beneficial, since among other things you'll end up with kind of a hybrid of traditional django templates and JSX-like custom html tags. I thought I'd share, in case anyone is interested in this.

**Comment by JuroOravec** [comment]
@oliverhaas I've moved this to an issue https://github.com/EmilStenstrom/django-components/issues/794

**Reply by oliverhaas** [reply]
Just that we're clear on this: I personally am not using this syntax, since I'm not sure if it's better to use the normal one and rather work on improving that (e.g. intellisense and formatters). So it's not really a feature request from my end or anything, that's why I posted it here.

**Reply by JuroOravec** [reply]
Fair. I do want to implement that, though. django-cotton grew quite fast, so I assume that there are people who prefer that syntax, and I want to make django-components accessible to them too.

**Reply by oliverhaas** [reply]
Sounds great! 
Yeah, I noticed that people really jumped on django-cotton. If the maintainer didn't say he wanted to stay away from python logic, I'm pretty sure at least one of my clients' projects would be using it right now.

**Comment by JuroOravec** [comment]
The HTML-like syntax is now part of [Citry](https://citry.dev/) ([citry-dev/citry](https://github.com/citry-dev/citry)).

For migration from django-components to Citry, see https://github.com/joeyjurjens/citry-django-poc
